src/pkg.cc

```diff
@@ -57,6 +59,126 @@ UInt T_SEMI   = 0;
 UInt T_BIPART = 0;
 UInt T_BLOCKS = 0;
 
+// Forward decl
+Obj FuncIMAGE_SET_TRANS_INT(Obj, Obj, Obj);
+
+static inline Int IS_MUTABLE_PLIST(Obj list) {
+  return !((TNUM_OBJ(list) - T_PLIST) % 2);
+}
```

Comment:
That function is a bad idea, it will break in future GAP versions, as encoding the (im)mutability in TNUMs has a good chance of going away at some point.

You can avoid this by conditionally compiling this code only when GAP is new enough, e.g using an `#if GAP_KERNEL_API_VERSION >= ...` statement. See also https://github.com/gap-system/gap/pull/3700 which I made just for that purpose :-)

Reply:
I know, I wasn't going to leave this in, I was just trying to make this work.

Reply:
Ah OK, sorry then -- and: excellent :-). Of course it's important to first get it to work (and thanks for working at this at all, much appreciated)